exercism/hamming/hamming.js

```diff
@@ -0,0 +1,16 @@
+class Hamming {
+  compute(hamming1, hamming2) {
+    if (hamming1.length !== hamming2.length) {
+      throw new Error('DNA strands must be of equal length.')
+    }
+    let similar = 0
+    for (let i = 0; i < hamming1.length; i++) {
+      if (hamming1[i] !== hamming2[i]) {
+      similar += 1
+
+      }
+    }
+    return similar
+  }
+}
+module.exports = Hamming;
```

Comment:
You could re-write this function using reduce instead of a for loop. You are using a mutable counter outside of the loop, where-as reduce is passing the result from function to function and never mutating a variable in its parent scope:
```js
return hamming1.split('').reduce( function(similarCounter, currentCharacter, index) {
  return currentCharacter !== hamming2[index] ? similarCounter + 1 : similarCounter
}, 0)
```